sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -1093,14 +1090,7 @@ public void processElement(ProcessContext c) throws IOException, SizeLimitExceed
         }
 
         // NOTE: The record id is always null.
-        output.add(
-            OutgoingMessage.of(
-                com.google.pubsub.v1.PubsubMessage.newBuilder()
-                    .setData(ByteString.copyFrom(payload))
-                    .putAllAttributes(attributes)
-                    .build(),
-                c.timestamp().getMillis(),
-                null));
+        output.add(OutgoingMessage.of(message, c.timestamp().getMillis(), null));
```

Comment:
Note that this only affects the bounded case (usually batch pipelines).

Reply:
The unbounded sink already uses this static method, where the com.google.pubsub.v1.PubsubMessage is being created from the PubsubIO PubsubMessage